p2p/module.go

```diff
@@ -21,7 +34,7 @@ import (
 var _ modules.P2PModule = &p2pModule{}
 
 type p2pModule struct {
-	bus modules.Bus
+	base_modules.IntegratableModule
```

Comment:
I am not sure I understand what has to be documented. If how to create modules is not clear I guess we should immediately have a chat and fix it.

We are applying SOLID principles

Reply:
We are indeed applying SOLID principles, and I just want to make it clearer (easier to understand, easier to navigate and easier to find) for other contributors. For example, I'm finding others (e.g. Bryan, h5law, etc..) finding it hard to understand which patterns to follow without my direct guidance.

----

As a new contributor to the codebase, one of the things I'd want to do is: create a new module.

1. Go to https://github.com/pokt-network/pocket/wiki/Modules-Readme
2. Search for "Module"
3. Find myself here: https://github.com/pokt-network/pocket/wiki/Modules-Readme
   - **Request 1**: Add a `TODO` to update that documentation. I reference the word `DOCUMENT` because that's the keyword we have in the `Makefile` for documentation specific TODOs
4. Eventually, since the docs are outdated, I might find myself looking at the structures below. In addition to the README, having a one liner could really help to anyone new reading the code

```
// A module that can be initialized with the bus and configured with a set of options
type InitializableModule interface { ... } 

// A module that can be registered and accessed via the bus
type IntegratableModule interface { ... } 

// A module with a lifecycle that can be started and stopped
type InterruptableModule interface { ... } 
}
```

Does that make sense?

Reply:
#509